src/App.tsx

```diff
@@ -1,7 +1,39 @@
-import React from 'react';
+import { Component, ReactNode } from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressedKey: string;
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    pressedKey: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.buttonClick);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.buttonClick);
+  }
+
+  buttonClick = (event: KeyboardEvent) => {
+    if (event.key) {
+      this.setState({ pressedKey: event.key });
+    }
+  };
+
```

Comment:
```suggestion

  buttonClick = (event: KeyboardEvent) => {
      this.setState({ pressedKey: event.key });
  };

```
`if` unnecessary here